build.gradle

```diff
@@ -606,6 +613,101 @@ project(':iceberg-dell') {
   }
 }
 
+project(':iceberg-azure') {
+  apply plugin: "com.bmuschko.docker-remote-api"
+  apply plugin: "com.github.johnrengelman.shadow"
+
+  dependencies {
+    api project(':iceberg-api')
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    implementation platform('com.azure:azure-sdk-bom')
```

Comment:
I would rather track dependency versions specifically rather than relying on some external BOM. BOMs are fine for end uses, but libraries should generally not delegate dependency versions to other projects.

Reply:
Ack. 
Will switch to direct dependencies.

Note: Currently GCP also relies on bom. We should also make a change in the GCP module to avoid using bom.

https://github.com/apache/iceberg/blob/7c2ea0128133f630041ecb77de1fae754073e904/build.gradle#L375

Reply:
If required, I can file another PR to eliminate the usage of bom in the GCP module.